Add rule docs for 0.16.1 integration updates (#1587)
elastic/security-docs: docs/detections/prebuilt-rules/downloadable-packages/0-16-1/prebuilt-rule-0-16-1-spike-in-aws-error-messages.asciidoc

=== Spike in AWS Error Messages

A machine learning job detected a significant spike in the rate of a particular error in the CloudTrail messages. Spikes in error messages may accompany attempts at privilege escalation, lateral movement, or discovery.

*Rule type*: machine_learning

*Rule indices*: None

*Severity*: low

*Risk score*: 21

*Runs every*: 15m

*Searches indices from*: now-60m ({ref}/common-options.html#date-math[Date Math format], see also <<rule-schedule, `Additional look-back time`>>)

*Maximum alerts per execution*: 100

*References*: 

* https://www.elastic.co/guide/en/security/current/prebuilt-ml-jobs.html

*Tags*: 

* Elastic
* Cloud
* AWS
* ML

*Version*: 8

*Rule authors*: 

* Elastic

*Rule license*: Elastic License v2

==== Investigation guide

[source, markdown]
----------------------------------
## Config

The AWS Fleet integration, Filebeat module, or similarly structured data is required to be compatible with this rule.

## Triage and analysis

### Investigating Spikes in CloudTrail Errors

CloudTrail logging provides visibility on actions taken within an AWS environment. By monitoring these events and understanding

> what is considered normal behavior within an organization, suspicious or malicious activity can be spotted when deviations

Reviewer: ```suggestion
what is considered normal behavior within an organization, you can spot suspicious or malicious activity when deviations
```

> are observed. This example rule triggers from a large spike in the number of CloudTrail log messages that contain a

Reviewer: ```suggestion
occur. This example rule triggers from a large spike in the number of CloudTrail log messages that contain a
```

- Consider the source IP address and geolocation for the calling user who issued the command. Do they look normal for the calling user? If the source is an EC2 IP address, is it associated with an EC2 instance in one of your accounts, or could it be sourcing from an EC2 instance that's not under your control? If it is an authorized EC2 instance, is the activity associated with normal behavior for the instance role or roles? Are there any other alerts or signs of suspicious activity involving this instance?

### False Positive Analysis
- This rule has the possibility to produce false positives based on unexpected activity occurring such as bugs or recent
changes to automation modules or scripting.
- The adoption of new services or the addition of new functionality to scripts may generate false positives.

### Related Rules
- Unusual AWS Command for a User
- Rare AWS Error Code

### Response and Remediation

> - If suspicious or malicious activity is observed, immediately rotate and delete relevant AWS IAM access keys

Reviewer: ```suggestion
- If suspicious or malicious activity is observed, immediately rotate and delete relevant AWS IAM access keys.
```

> - Validate if any unauthorized new users were created, remove these accounts and request password resets for other IAM users

Reviewer: ```suggestion
- If any unauthorized new user accounts were created, remove them. Request password resets for other IAM users.
```

> - Look into enabling multi-factor authentication for users

Reviewer: ```suggestion
- Look into enabling multi-factor authentication for users.
```

> - Follow security best practices [outlined](https://aws.amazon.com/premiumsupport/knowledge-center/security-best-practices/) by AWS

Reviewer: ```suggestion
- Follow security best practices [outlined](https://aws.amazon.com/premiumsupport/knowledge-center/security-best-practices/) by AWS.
```
